// unocss-preset-vuetify/dist/index.cjs
'use strict';

function preflights(options) {
  if (!options.font) {
    return void 0;
  }
  return [
    {
      layer: "typography",
      getCSS() {
        if (options.font) {
          const variables = Object.entries(options.font).map(([key, value]) => `--v-font-${key}: ${value};`).join("\n");
          return `:root {
${variables}
}`;
        } else {
          return "";
        }
      }
    }
  ];
}

const defaultBreakpoints = {
  sm: "600px",
  md: "960px",
  lg: "1280px",
  xl: "1920px",
  xxl: "2560px"
};
const defaultSpacer = 4;
const defaultSpacerSteps = 16;
const fontWeights = {
  thin: 100,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  black: 900
};
const borderRadii = {
  "0": "0",
  "sm": "2px",
  "": "4px",
  "lg": "8px",
  "xl": "24px",
  "pill": "9999px",
  "circle": "50%",
  "shaped": "24px 0"
};
const typographyPresets = {
  md2: {
    h1: { fontFamily: "var(--v-font-heading)", fontSize: "6rem", fontWeight: 300, lineHeight: 1.167, letterSpacing: "-0.015625em" },
    h2: { fontFamily: "var(--v-font-heading)", fontSize: "3.75rem", fontWeight: 300, lineHeight: 1.2, letterSpacing: "-0.0083333333em" },
    h3: { fontFamily: "var(--v-font-heading)", fontSize: "3rem", fontWeight: 400, lineHeight: 1.167, letterSpacing: "normal" },
    h4: { fontFamily: "var(--v-font-heading)", fontSize: "2.125rem", fontWeight: 400, lineHeight: 1.235, letterSpacing: "0.0073529412em" },
    h5: { fontFamily: "var(--v-font-heading)", fontSize: "1.5rem", fontWeight: 400, lineHeight: 1.334, letterSpacing: "normal" },
    h6: { fontFamily: "var(--v-font-heading)", fontSize: "1.25rem", fontWeight: 500, lineHeight: 1.6, letterSpacing: "0.0125em" },
    subtitle1: { fontFamily: "var(--v-font-body)", fontSize: "1rem", fontWeight: 400, lineHeight: 1.75, letterSpacing: "0.009375em" },
    subtitle2: { fontFamily: "var(--v-font-body)", fontSize: "0.875rem", fontWeight: 500, lineHeight: 1.57, letterSpacing: "0.0071428571em" },
    body1: { fontFamily: "var(--v-font-body)", fontSize: "1rem", fontWeight: 400, lineHeight: 1.5, letterSpacing: "0.03125em" },
    body2: { fontFamily: "var(--v-font-body)", fontSize: "0.875rem", fontWeight: 400, lineHeight: 1.43, letterSpacing: "0.0178571429em" },
    button: { fontFamily: "var(--v-font-body)", fontSize: "0.875rem", fontWeight: 500, lineHeight: 1.75, letterSpacing: "0.0892857143em", textTransform: "uppercase" },
    caption: { fontFamily: "var(--v-font-body)", fontSize: "0.75rem", fontWeight: 400, lineHeight: 1.66, letterSpacing: "0.0333333333em" },
    overline: { fontFamily: "var(--v-font-body)", fontSize: "0.75rem", fontWeight: 500, lineHeight: 2.66, letterSpacing: "0.1666666667em", textTransform: "uppercase" }
  },
  md3: {
    "display-large": { fontFamily: "var(--v-font-heading)", fontSize: "3.5625rem", fontWeight: 400, lineHeight: 1.1228070175, letterSpacing: "-0.0043859649em" },
    "display-medium": { fontFamily: "var(--v-font-heading)", fontSize: "2.8125rem", fontWeight: 400, lineHeight: 1.1555555556, letterSpacing: "normal" },
    "display-small": { fontFamily: "var(--v-font-heading)", fontSize: "2.25rem", fontWeight: 400, lineHeight: 1.2222222222, letterSpacing: "normal" },
    "headline-large": { fontFamily: "var(--v-font-heading)", fontSize: "2rem", fontWeight: 400, lineHeight: 1.25, letterSpacing: "normal" },
    "headline-medium": { fontFamily: "var(--v-font-heading)", fontSize: "1.75rem", fontWeight: 400, lineHeight: 1.2857142857, letterSpacing: "normal" },
    "headline-small": { fontFamily: "var(--v-font-heading)", fontSize: "1.5rem", fontWeight: 400, lineHeight: 1.3333333333, letterSpacing: "normal" },
    "title-large": { fontFamily: "var(--v-font-heading)", fontSize: "1.375rem", fontWeight: 400, lineHeight: 1.2727272727, letterSpacing: "normal" },
    "title-medium": { fontFamily: "var(--v-font-body)", fontSize: "1rem", fontWeight: 500, lineHeight: 1.5, letterSpacing: "0.009375em" },
    "title-small": { fontFamily: "var(--v-font-body)", fontSize: "0.875rem", fontWeight: 500, lineHeight: 1.4285714286, letterSpacing: "0.0071428571em" },
    "body-large": { fontFamily: "var(--v-font-body)", fontSize: "1rem", fontWeight: 400, lineHeight: 1.5, letterSpacing: "0.03125em" },
    "body-medium": { fontFamily: "var(--v-font-body)", fontSize: "0.875rem", fontWeight: 400, lineHeight: 1.4285714286, letterSpacing: "0.0178571429em" },
    "body-small": { fontFamily: "var(--v-font-body)", fontSize: "0.75rem", fontWeight: 400, lineHeight: 1.3333333333, letterSpacing: "0.0333333333em" },
    "label-large": { fontFamily: "var(--v-font-body)", fontSize: "0.875rem", fontWeight: 500, lineHeight: 1.4285714286, letterSpacing: "0.0071428571em" },
    "label-medium": { fontFamily: "var(--v-font-body)", fontSize: "0.75rem", fontWeight: 500, lineHeight: 1.3333333333, letterSpacing: "0.0416666667em" },
    "label-small": { fontFamily: "var(--v-font-body)", fontSize: "0.6875rem", fontWeight: 500, lineHeight: 1.4545454545, letterSpacing: "0.0454545455em" }
  }
};
const elevationOverlayStep = 2;
const elevationPresets = {
  md2: Object.fromEntries(
    [
      "0px  0px  0px  0px [1], 0px  0px  0px 0px [2], 0px 0px  0px 0px [3]",
      "0px  2px  1px -1px [1], 0px  1px  1px 0px [2], 0px 1px  3px 0px [3]",
      "0px  3px  1px -2px [1], 0px  2px  2px 0px [2], 0px 1px  5px 0px [3]",
      "0px  3px  3px -2px [1], 0px  3px  4px 0px [2], 0px 1px  8px 0px [3]",
      "0px  2px  4px -1px [1], 0px  4px  5px 0px [2], 0px 1px 10px 0px [3]",
      "0px  3px  5px -1px [1], 0px  5px  8px 0px [2], 0px 1px 14px 0px [3]",
      "0px  3px  5px -1px [1], 0px  6px 10px 0px [2], 0px 1px 18px 0px [3]",
      "0px  4px  5px -2px [1], 0px  7px 10px 1px [2], 0px 2px 16px 1px [3]",
      "0px  5px  5px -3px [1], 0px  8px 10px 1px [2], 0px 3px 14px 2px [3]",
      "0px  5px  6px -3px [1], 0px  9px 12px 1px [2], 0px 3px 16px 2px [3]",
      "0px  6px  6px -3px [1], 0px 10px 14px 1px [2], 0px 4px 18px 3px [3]",
      "0px  6px  7px -4px [1], 0px 11px 15px 1px [2], 0px 4px 20px 3px [3]",
      "0px  7px  8px -4px [1], 0px 12px 17px 2px [2], 0px 5px 22px 4px [3]",
      "0px  7px  8px -4px [1], 0px 13px 19px 2px [2], 0px 5px 24px 4px [3]",
      "0px  7px  9px -4px [1], 0px 14px 21px 2px [2], 0px 5px 26px 4px [3]",
      "0px  8px  9px -5px [1], 0px 15px 22px 2px [2], 0px 6px 28px 5px [3]",
      "0px  8px 10px -5px [1], 0px 16px 24px 2px [2], 0px 6px 30px 5px [3]",
      "0px  8px 11px -5px [1], 0px 17px 26px 2px [2], 0px 6px 32px 5px [3]",
      "0px  9px 11px -5px [1], 0px 18px 28px 2px [2], 0px 7px 34px 6px [3]",
      "0px  9px 12px -6px [1], 0px 19px 29px 2px [2], 0px 7px 36px 6px [3]",
      "0px 10px 13px -6px [1], 0px 20px 31px 3px [2], 0px 8px 38px 7px [3]",
      "0px 10px 13px -6px [1], 0px 21px 33px 3px [2], 0px 8px 40px 7px [3]",
      "0px 10px 14px -6px [1], 0px 22px 35px 3px [2], 0px 8px 42px 7px [3]",
      "0px 11px 14px -7px [1], 0px 23px 36px 3px [2], 0px 9px 44px 8px [3]",
      "0px 11px 15px -7px [1], 0px 24px 38px 3px [2], 0px 9px 46px 8px [3]"
    ].map(
      (line) => line.replace("[1]", "var(--v-shadow-key-umbra-opacity,    rgba(0, 0, 0, 0.2 ))").replace("[2]", "var(--v-shadow-key-penumbra-opacity, rgba(0, 0, 0, 0.14))").replace("[3]", "var(--v-shadow-key-ambient-opacity,  rgba(0, 0, 0, 0.12))")
    ).map((line, i) => [String(i), { "box-shadow": line }])
  ),
  md3: Object.fromEntries(
    [
      "0px 0px 0px 0px [1], 0px 0px  0px 0px [2]",
      "0px 1px 2px 0px [1], 0px 1px  3px 1px [2]",
      "0px 1px 2px 0px [1], 0px 2px  6px 2px [2]",
      "0px 1px 3px 0px [1], 0px 4px  8px 3px [2]",
      "0px 2px 3px 0px [1], 0px 6px 10px 4px [2]",
      "0px 4px 4px 0px [1], 0px 8px 12px 6px [2]"
    ].map(
      (line) => line.replace("[1]", "rgba(var(--v-shadow-color), var(--v-shadow-key-opacity,     0.3 ))").replace("[2]", "rgba(var(--v-shadow-color), var(--v-shadow-ambient-opacity, 0.15))")
    ).map((line, i) => [String(i), {
      "box-shadow": line,
      "--v-elevation-overlay": `color-mix(in srgb, var(--v-elevation-overlay-color, #000) ${i * elevationOverlayStep}%, transparent)`
    }])
  )
};

const borderWidths = {
  0: "0",
  thin: "thin",
  sm: "1px",
  md: "2px",
  lg: "4px",
  xl: "8px"
};
function borderRules() {
  const rules = [["border", { "border-style": "solid", "border-width": "thin" }]];
  for (const [name, width] of Object.entries(borderWidths)) {
    rules.push([`border-${name}`, { "border-width": width }]);
  }
  const borderDirections = {
    t: "border-top",
    b: "border-bottom",
    l: "border-left",
    r: "border-right"
  };
  for (const [dir, prop] of Object.entries(borderDirections)) {
    rules.push([`border-${dir}`, { [`${prop}-style`]: "solid", [`${prop}-width`]: "thin" }]);
    for (const [name, width] of Object.entries(borderWidths)) {
      rules.push([`border-${dir}-${name}`, { [`${prop}-width`]: width }]);
    }
  }
  rules.push(["border-s", { "border-inline-start-style": "solid", "border-inline-start-width": "thin" }], ["border-e", { "border-inline-end-style": "solid", "border-inline-end-width": "thin" }]);
  for (const [name, width] of Object.entries(borderWidths)) {
    rules.push([`border-s-${name}`, { "border-inline-start-width": width }], [`border-e-${name}`, { "border-inline-end-width": width }]);
  }
  const borderStyles = ["solid", "dashed", "dotted", "double", "none"];
  for (const style of borderStyles) {
    rules.push([`border-${style}`, { "border-style": style }]);
  }
  for (const opacity of [0, 25, 50, 75, 100]) {
    rules.push([`border-opacity-${opacity}`, { "--v-border-opacity": String(opacity / 100) }]);
  }
  rules.push(["border-current", { "border-color": "currentColor" }]);
  for (const [name, value] of Object.entries(borderRadii)) {
    const className = name === "" ? "rounded" : `rounded-${name}`;
    rules.push([className, { "border-radius": value }]);
  }
  const radiusDirections = {
    t: ["border-top-left-radius", "border-top-right-radius"],
    b: ["border-bottom-left-radius", "border-bottom-right-radius"],
    l: ["border-top-left-radius", "border-bottom-left-radius"],
    r: ["border-top-right-radius", "border-bottom-right-radius"],
    tl: ["border-top-left-radius"],
    tr: ["border-top-right-radius"],
    bl: ["border-bottom-left-radius"],
    br: ["border-bottom-right-radius"]
  };
  for (const [dir, props] of Object.entries(radiusDirections)) {
    for (const [name, value] of Object.entries(borderRadii)) {
      const className = name === "" ? `rounded-${dir}` : `rounded-${dir}-${name}`;
      const css = Object.fromEntries(props.map((p) => [p, value]));
      rules.push([className, css]);
    }
  }
  const logicalRadiusDirections = {
    s: ["border-start-start-radius", "border-end-start-radius"],
    e: ["border-start-end-radius", "border-end-end-radius"],
    ts: ["border-start-start-radius"],
    te: ["border-start-end-radius"],
    bs: ["border-end-start-radius"],
    be: ["border-end-end-radius"]
  };
  for (const [dir, props] of Object.entries(logicalRadiusDirections)) {
    for (const [name, value] of Object.entries(borderRadii)) {
      const className = name === "" ? `rounded-${dir}` : `rounded-${dir}-${name}`;
      const css = Object.fromEntries(props.map((p) => [p, value]));
      rules.push([className, css]);
    }
  }
  return rules;
}

const mainTRC = 2.4;
const Rco = 0.2126729;
const Gco = 0.7151522;
const Bco = 0.072175;
const normBG = 0.55;
const normTXT = 0.58;
const revTXT = 0.57;
const revBG = 0.62;
const blkThrs = 0.03;
const blkClmp = 1.45;
const deltaYmin = 5e-4;
const scaleBoW = 1.25;
const scaleWoB = 1.25;
const loConThresh = 0.078;
const loConFactor = 12.82051282051282;
const loConOffset = 0.06;
const loClip = 1e-3;
function APCAcontrast(text, background) {
  const Rtxt = (text.r / 255) ** mainTRC;
  const Gtxt = (text.g / 255) ** mainTRC;
  const Btxt = (text.b / 255) ** mainTRC;
  const Rbg = (background.r / 255) ** mainTRC;
  const Gbg = (background.g / 255) ** mainTRC;
  const Bbg = (background.b / 255) ** mainTRC;
  let Ytxt = Rtxt * Rco + Gtxt * Gco + Btxt * Bco;
  let Ybg = Rbg * Rco + Gbg * Gco + Bbg * Bco;
  if (Ytxt <= blkThrs) {
    Ytxt += (blkThrs - Ytxt) ** blkClmp;
  }
  if (Ybg <= blkThrs) {
    Ybg += (blkThrs - Ybg) ** blkClmp;
  }
  if (Math.abs(Ybg - Ytxt) < deltaYmin) {
    return 0;
  }
  let outputContrast;
  if (Ybg > Ytxt) {
    const SAPC = (Ybg ** normBG - Ytxt ** normTXT) * scaleBoW;
    outputContrast = SAPC < loClip ? 0 : SAPC < loConThresh ? SAPC - SAPC * loConFactor * loConOffset : SAPC - loConOffset;
  } else {
    const SAPC = (Ybg ** revBG - Ytxt ** revTXT) * scaleWoB;
    outputContrast = SAPC > -loClip ? 0 : SAPC > -loConThresh ? SAPC - SAPC * loConFactor * loConOffset : SAPC + loConOffset;
  }
  return outputContrast * 100;
}

const shades = {
  black: "#000000",
  white: "#FFFFFF",
  transparent: "transparent"
};
const paletteColors = {
  "red": {
    "base": "#F44336",
    "lighten-5": "#FFEBEE",
    "lighten-4": "#FFCDD2",
    "lighten-3": "#EF9A9A",
    "lighten-2": "#E57373",
    "lighten-1": "#EF5350",
    "darken-1": "#E53935",
    "darken-2": "#D32F2F",
    "darken-3": "#C62828",
    "darken-4": "#B71C1C",
    "accent-1": "#FF8A80",
    "accent-2": "#FF5252",
    "accent-3": "#FF1744",
    "accent-4": "#D50000"
  },
  "pink": {
    "base": "#e91e63",
    "lighten-5": "#fce4ec",
    "lighten-4": "#f8bbd0",
    "lighten-3": "#f48fb1",
    "lighten-2": "#f06292",
    "lighten-1": "#ec407a",
    "darken-1": "#d81b60",
    "darken-2": "#c2185b",
    "darken-3": "#ad1457",
    "darken-4": "#880e4f",
    "accent-1": "#ff80ab",
    "accent-2": "#ff4081",
    "accent-3": "#f50057",
    "accent-4": "#c51162"
  },
  "purple": {
    "base": "#9c27b0",
    "lighten-5": "#f3e5f5",
    "lighten-4": "#e1bee7",
    "lighten-3": "#ce93d8",
    "lighten-2": "#ba68c8",
    "lighten-1": "#ab47bc",
    "darken-1": "#8e24aa",
    "darken-2": "#7b1fa2",
    "darken-3": "#6a1b9a",
    "darken-4": "#4a148c",
    "accent-1": "#ea80fc",
    "accent-2": "#e040fb",
    "accent-3": "#d500f9",
    "accent-4": "#aa00ff"
  },
  "deep-purple": {
    "base": "#673ab7",
    "lighten-5": "#ede7f6",
    "lighten-4": "#d1c4e9",
    "lighten-3": "#b39ddb",
    "lighten-2": "#9575cd",
    "lighten-1": "#7e57c2",
    "darken-1": "#5e35b1",
    "darken-2": "#512da8",
    "darken-3": "#4527a0",
    "darken-4": "#311b92",
    "accent-1": "#b388ff",
    "accent-2": "#7c4dff",
    "accent-3": "#651fff",
    "accent-4": "#6200ea"
  },
  "indigo": {
    "base": "#3f51b5",
    "lighten-5": "#e8eaf6",
    "lighten-4": "#c5cae9",
    "lighten-3": "#9fa8da",
    "lighten-2": "#7986cb",
    "lighten-1": "#5c6bc0",
    "darken-1": "#3949ab",
    "darken-2": "#303f9f",
    "darken-3": "#283593",
    "darken-4": "#1a237e",
    "accent-1": "#8c9eff",
    "accent-2": "#536dfe",
    "accent-3": "#3d5afe",
    "accent-4": "#304ffe"
  },
  "blue": {
    "base": "#2196F3",
    "lighten-5": "#E3F2FD",
    "lighten-4": "#BBDEFB",
    "lighten-3": "#90CAF9",
    "lighten-2": "#64B5F6",
    "lighten-1": "#42A5F5",
    "darken-1": "#1E88E5",
    "darken-2": "#1976D2",
    "darken-3": "#1565C0",
    "darken-4": "#0D47A1",
    "accent-1": "#82B1FF",
    "accent-2": "#448AFF",
    "accent-3": "#2979FF",
    "accent-4": "#2962FF"
  },
  "light-blue": {
    "base": "#03a9f4",
    "lighten-5": "#e1f5fe",
    "lighten-4": "#b3e5fc",
    "lighten-3": "#81d4fa",
    "lighten-2": "#4fc3f7",
    "lighten-1": "#29b6f6",
    "darken-1": "#039be5",
    "darken-2": "#0288d1",
    "darken-3": "#0277bd",
    "darken-4": "#01579b",
    "accent-1": "#80d8ff",
    "accent-2": "#40c4ff",
    "accent-3": "#00b0ff",
    "accent-4": "#0091ea"
  },
  "cyan": {
    "base": "#00bcd4",
    "lighten-5": "#e0f7fa",
    "lighten-4": "#b2ebf2",
    "lighten-3": "#80deea",
    "lighten-2": "#4dd0e1",
    "lighten-1": "#26c6da",
    "darken-1": "#00acc1",
    "darken-2": "#0097a7",
    "darken-3": "#00838f",
    "darken-4": "#006064",
    "accent-1": "#84ffff",
    "accent-2": "#18ffff",
    "accent-3": "#00e5ff",
    "accent-4": "#00b8d4"
  },
  "teal": {
    "base": "#009688",
    "lighten-5": "#e0f2f1",
    "lighten-4": "#b2dfdb",
    "lighten-3": "#80cbc4",
    "lighten-2": "#4db6ac",
    "lighten-1": "#26a69a",
    "darken-1": "#00897b",
    "darken-2": "#00796b",
    "darken-3": "#00695c",
    "darken-4": "#004d40",
    "accent-1": "#a7ffeb",
    "accent-2": "#64ffda",
    "accent-3": "#1de9b6",
    "accent-4": "#00bfa5"
  },
  "green": {
    "base": "#4CAF50",
    "lighten-5": "#E8F5E9",
    "lighten-4": "#C8E6C9",
    "lighten-3": "#A5D6A7",
    "lighten-2": "#81C784",
    "lighten-1": "#66BB6A",
    "darken-1": "#43A047",
    "darken-2": "#388E3C",
    "darken-3": "#2E7D32",
    "darken-4": "#1B5E20",
    "accent-1": "#B9F6CA",
    "accent-2": "#69F0AE",
    "accent-3": "#00E676",
    "accent-4": "#00C853"
  },
  "light-green": {
    "base": "#8bc34a",
    "lighten-5": "#f1f8e9",
    "lighten-4": "#dcedc8",
    "lighten-3": "#c5e1a5",
    "lighten-2": "#aed581",
    "lighten-1": "#9ccc65",
    "darken-1": "#7cb342",
    "darken-2": "#689f38",
    "darken-3": "#558b2f",
    "darken-4": "#33691e",
    "accent-1": "#ccff90",
    "accent-2": "#b2ff59",
    "accent-3": "#76ff03",
    "accent-4": "#64dd17"
  },
  "lime": {
    "base": "#cddc39",
    "lighten-5": "#f9fbe7",
    "lighten-4": "#f0f4c3",
    "lighten-3": "#e6ee9c",
    "lighten-2": "#dce775",
    "lighten-1": "#d4e157",
    "darken-1": "#c0ca33",
    "darken-2": "#afb42b",
    "darken-3": "#9e9d24",
    "darken-4": "#827717",
    "accent-1": "#f4ff81",
    "accent-2": "#eeff41",
    "accent-3": "#c6ff00",
    "accent-4": "#aeea00"
  },
  "yellow": {
    "base": "#ffeb3b",
    "lighten-5": "#fffde7",
    "lighten-4": "#fff9c4",
    "lighten-3": "#fff59d",
    "lighten-2": "#fff176",
    "lighten-1": "#ffee58",
    "darken-1": "#fdd835",
    "darken-2": "#fbc02d",
    "darken-3": "#f9a825",
    "darken-4": "#f57f17",
    "accent-1": "#ffff8d",
    "accent-2": "#ffff00",
    "accent-3": "#ffea00",
    "accent-4": "#ffd600"
  },
  "amber": {
    "base": "#ffc107",
    "lighten-5": "#fff8e1",
    "lighten-4": "#ffecb3",
    "lighten-3": "#ffe082",
    "lighten-2": "#ffd54f",
    "lighten-1": "#ffca28",
    "darken-1": "#ffb300",
    "darken-2": "#ffa000",
    "darken-3": "#ff8f00",
    "darken-4": "#ff6f00",
    "accent-1": "#ffe57f",
    "accent-2": "#ffd740",
    "accent-3": "#ffc400",
    "accent-4": "#ffab00"
  },
  "orange": {
    "base": "#ff9800",
    "lighten-5": "#fff3e0",
    "lighten-4": "#ffe0b2",
    "lighten-3": "#ffcc80",
    "lighten-2": "#ffb74d",
    "lighten-1": "#ffa726",
    "darken-1": "#fb8c00",
    "darken-2": "#f57c00",
    "darken-3": "#ef6c00",
    "darken-4": "#e65100",
    "accent-1": "#ffd180",
    "accent-2": "#ffab40",
    "accent-3": "#ff9100",
    "accent-4": "#ff6d00"
  },
  "deep-orange": {
    "base": "#ff5722",
    "lighten-5": "#fbe9e7",
    "lighten-4": "#ffccbc",
    "lighten-3": "#ffab91",
    "lighten-2": "#ff8a65",
    "lighten-1": "#ff7043",
    "darken-1": "#f4511e",
    "darken-2": "#e64a19",
    "darken-3": "#d84315",
    "darken-4": "#bf360c",
    "accent-1": "#ff9e80",
    "accent-2": "#ff6e40",
    "accent-3": "#ff3d00",
    "accent-4": "#dd2c00"
  },
  "brown": {
    "base": "#795548",
    "lighten-5": "#efebe9",
    "lighten-4": "#d7ccc8",
    "lighten-3": "#bcaaa4",
    "lighten-2": "#a1887f",
    "lighten-1": "#8d6e63",
    "darken-1": "#6d4c41",
    "darken-2": "#5d4037",
    "darken-3": "#4e342e",
    "darken-4": "#3e2723"
  },
  "blue-grey": {
    "base": "#607d8b",
    "lighten-5": "#eceff1",
    "lighten-4": "#cfd8dc",
    "lighten-3": "#b0bec5",
    "lighten-2": "#90a4ae",
    "lighten-1": "#78909c",
    "darken-1": "#546e7a",
    "darken-2": "#455a64",
    "darken-3": "#37474f",
    "darken-4": "#263238"
  },
  "grey": {
    "base": "#9e9e9e",
    "lighten-5": "#fafafa",
    "lighten-4": "#f5f5f5",
    "lighten-3": "#eeeeee",
    "lighten-2": "#e0e0e0",
    "lighten-1": "#bdbdbd",
    "darken-1": "#757575",
    "darken-2": "#616161",
    "darken-3": "#424242",
    "darken-4": "#212121"
  }
};

function padEnd(str, length, char = "0") {
  return str + char.repeat(Math.max(0, length - str.length));
}
function chunk(str, size = 1) {
  const chunked = [];
  let index = 0;
  while (index < str.length) {
    chunked.push(str.slice(index, index + size));
    index += size;
  }
  return chunked;
}
function parseHex(hex) {
  if (hex.startsWith("#")) {
    hex = hex.slice(1);
  }
  hex = hex.replace(/([^0-9a-f])/gi, "F");
  if (hex.length === 3 || hex.length === 4) {
    hex = hex.split("").map((x) => x + x).join("");
  }
  if (hex.length !== 6) {
    hex = padEnd(padEnd(hex, 6), 8, "F");
  }
  return hex;
}
function HexToRGB(hex) {
  hex = parseHex(hex);
  const [r, g, b, a] = chunk(hex, 2).map((c) => Number.parseInt(c, 16));
  return { r, g, b, a: a === void 0 ? a : a / 255 };
}
function parseColor(color) {
  if (typeof color === "number") {
    if (Number.isNaN(color) || color < 0 || color > 16777215) {
      console.warn(`'${color}' is not a valid hex color`);
    }
    return {
      r: (color & 16711680) >> 16,
      g: (color & 65280) >> 8,
      b: color & 255
    };
  } else if (typeof color === "string" && color.startsWith("#")) {
    let hex = color.slice(1);
    if ([3, 4].includes(hex.length)) {
      hex = hex.split("").map((char) => char + char).join("");
    } else if (![6, 8].includes(hex.length)) {
      console.warn(`'${color}' is not a valid hex(a) color`);
    }
    const int = Number.parseInt(hex, 16);
    if (Number.isNaN(int) || int < 0 || int > 4294967295) {
      console.warn(`'${color}' is not a valid hex(a) color`);
    }
    return HexToRGB(hex);
  }
  throw new TypeError(`Invalid color: ${color == null ? color : String(color) || color.constructor.name}
Expected #hex or number`);
}

const themeColors = [
  "primary",
  "secondary",
  "success",
  "info",
  "warning",
  "error",
  "background",
  "surface",
  "surface-variant",
  "surface-bright",
  "on-primary",
  "on-secondary",
  "on-success",
  "on-info",
  "on-warning",
  "on-error",
  "on-background",
  "on-surface",
  "on-surface-variant"
];
function getForeground(color) {
  const blackContrast = Math.abs(APCAcontrast(parseColor(0), parseColor(color)));
  const whiteContrast = Math.abs(APCAcontrast(parseColor(16777215), parseColor(color)));
  return whiteContrast > Math.min(blackContrast, 50) ? "#fff" : "#000";
}
function bgCss(color) {
  return {
    "background-color": color,
    "color": getForeground(color)
  };
}
function textCss(color) {
  return { color };
}
function colorRules() {
  const rules = [];
  for (const name of themeColors) {
    rules.push(
      [`bg-${name}`, {
        "background-color": `rgb(var(--v-theme-${name}))`,
        "color": `rgba(var(--v-theme-on-${name}), var(--v-high-emphasis-opacity))`
      }],
      [`text-${name}`, {
        color: `rgb(var(--v-theme-${name}))`
      }]
    );
  }
  for (const [name, value] of Object.entries(shades)) {
    if (value === "transparent") {
      rules.push(
        [`bg-${name}`, { "background-color": value }],
        [`text-${name}`, textCss(value)]
      );
    } else {
      rules.push(
        [`bg-${name}`, bgCss(value)],
        [`text-${name}`, textCss(value)]
      );
    }
  }
  for (const [colorName, colorShades] of Object.entries(paletteColors)) {
    for (const [shade, value] of Object.entries(colorShades)) {
      const suffix = shade === "base" ? colorName : `${colorName}-${shade}`;
      rules.push(
        [`bg-${suffix}`, bgCss(value)],
        [`text-${suffix}`, textCss(value)]
      );
    }
  }
  return rules;
}

const cursorValues = [
  "auto",
  "default",
  "pointer",
  "wait",
  "text",
  "move",
  "help",
  "not-allowed",
  "progress",
  "grab",
  "grabbing",
  "none",
  "context-menu",
  "cell",
  "crosshair",
  "vertical-text",
  "alias",
  "copy",
  "no-drop",
  "all-scroll",
  "col-resize",
  "row-resize",
  "n-resize",
  "e-resize",
  "s-resize",
  "w-resize",
  "ne-resize",
  "nw-resize",
  "se-resize",
  "sw-resize",
  "ew-resize",
  "ns-resize",
  "nesw-resize",
  "nwse-resize",
  "zoom-in",
  "zoom-out"
];
function cursorRules() {
  const rules = [];
  for (const value of cursorValues) {
    rules.push([`cursor-${value}`, { cursor: value }]);
  }
  return rules;
}

const displayValues = [
  "none",
  "inline",
  "inline-block",
  "block",
  "table",
  "table-row",
  "table-cell",
  "flex",
  "inline-flex"
];
function displayRules() {
  const rules = [];
  for (const value of displayValues) {
    rules.push([`d-${value}`, { display: value }]);
  }
  for (const value of displayValues) {
    rules.push([
      `d-print-${value}`,
      { display: value },
      { layer: "utilities", custom: { print: true } }
    ]);
  }
  rules.push([
    "d-sr-only",
    {
      "position": "absolute",
      "width": "1px",
      "height": "1px",
      "padding": "0",
      "margin": "-1px",
      "overflow": "hidden",
      "clip": "rect(0, 0, 0, 0)",
      "white-space": "nowrap",
      "border": "0"
    }
  ], [
    "d-sr-only-focusable",
    {
      "position": "absolute",
      "width": "1px",
      "height": "1px",
      "padding": "0",
      "margin": "-1px",
      "overflow": "hidden",
      "clip": "rect(0, 0, 0, 0)",
      "white-space": "nowrap",
      "border": "0"
    },
    { custom: { focusable: true } }
  ]);
  return rules;
}

function spacerValue(n, spacer) {
  return `${n * spacer}px`;
}
const directions = {
  a: [""],
  t: ["-top"],
  b: ["-bottom"],
  l: ["-left"],
  r: ["-right"],
  x: ["-left", "-right"],
  y: ["-top", "-bottom"]
};
function kebabCase(str) {
  return (str ?? "").replace(/([a-z0-9])([A-Z])/g, "$1 $2").replace(/([A-Z]+)([A-Z][a-z])/g, "$1 $2").trim().split(/\s+/).map((p) => p.toLowerCase()).join("-");
}

function elevationRules(elevation = "md3") {
  const preset = elevation === "md2" ? elevationPresets.md2 : elevation === "md3" ? elevationPresets.md3 : elevation;
  const rules = [];
  for (const [level, styles] of Object.entries(preset)) {
    const css = Object.entries(styles).reduce(
      (o, [key, value]) => ({ ...o, [kebabCase(key)]: value }),
      {}
    );
    rules.push([`elevation-${level}`, css]);
  }
  rules.push(["elevation-overlay", { "background-image": "linear-gradient(var(--v-elevation-overlay), var(--v-elevation-overlay))" }]);
  return rules;
}

function flexRules() {
  const rules = [["flex-row", { "flex-direction": "row" }], ["flex-row-reverse", { "flex-direction": "row-reverse" }], ["flex-column", { "flex-direction": "column" }], ["flex-column-reverse", { "flex-direction": "column-reverse" }], ["flex-wrap", { "flex-wrap": "wrap" }], ["flex-nowrap", { "flex-wrap": "nowrap" }], ["flex-wrap-reverse", { "flex-wrap": "wrap-reverse" }], ["flex-grow-0", { "flex-grow": "0" }], ["flex-grow-1", { "flex-grow": "1" }], ["flex-shrink-0", { "flex-shrink": "0" }], ["flex-shrink-1", { "flex-shrink": "1" }], ["flex-fill", { flex: "1 1 auto" }], ["flex-1-1", { flex: "1 1 auto" }], ["flex-1-0", { flex: "1 0 auto" }], ["flex-0-1", { flex: "0 1 auto" }], ["flex-0-0", { flex: "0 0 auto" }], ["flex-1-1-0", { flex: "1 1 0%" }], ["flex-1-1-100", { flex: "1 1 100%" }], ["flex-1-0-0", { flex: "1 0 0%" }], ["flex-1-0-100", { flex: "1 0 100%" }], ["flex-0-1-0", { flex: "0 1 0%" }], ["flex-0-1-100", { flex: "0 1 100%" }], ["flex-0-0-0", { flex: "0 0 0%" }], ["flex-0-0-100", { flex: "0 0 100%" }], ["justify-start", { "justify-content": "flex-start" }], ["justify-end", { "justify-content": "flex-end" }], ["justify-center", { "justify-content": "center" }], ["justify-space-between", { "justify-content": "space-between" }], ["justify-space-around", { "justify-content": "space-around" }], ["justify-space-evenly", { "justify-content": "space-evenly" }], ["justify-items-start", { "justify-items": "start" }], ["justify-items-end", { "justify-items": "end" }], ["justify-items-center", { "justify-items": "center" }], ["justify-items-stretch", { "justify-items": "stretch" }], ["align-start", { "align-items": "flex-start" }], ["align-end", { "align-items": "flex-end" }], ["align-center", { "align-items": "center" }], ["align-baseline", { "align-items": "baseline" }], ["align-stretch", { "align-items": "stretch" }], ["align-content-start", { "align-content": "flex-start" }], ["align-content-end", { "align-content": "flex-end" }], ["align-content-center", { "align-content": "center" }], ["align-content-space-between", { "align-content": "space-between" }], ["align-content-space-around", { "align-content": "space-around" }], ["align-content-space-evenly", { "align-content": "space-evenly" }], ["align-content-stretch", { "align-content": "stretch" }], ["align-self-auto", { "align-self": "auto" }], ["align-self-start", { "align-self": "flex-start" }], ["align-self-end", { "align-self": "flex-end" }], ["align-self-center", { "align-self": "center" }], ["align-self-baseline", { "align-self": "baseline" }], ["align-self-stretch", { "align-self": "stretch" }], ["order-first", { order: "-1" }], ["order-last", { order: "13" }]];
  for (let i = 0; i <= 12; i++) {
    rules.push([`order-${i}`, { order: String(i) }]);
  }
  return rules;
}

function floatRules() {
  const rules = [["float-none", { float: "none" }], ["float-left", { float: "left" }], ["float-right", { float: "right" }], ["clear-none", { clear: "none" }], ["clear-left", { clear: "left" }], ["clear-right", { clear: "right" }], ["clear-both", { clear: "both" }]];
  return rules;
}

const COLS = 12;
function colPercent(n) {
  return `${+(n / COLS * 100).toFixed(10)}%`;
}
function gridRules() {
  const rules = [];
  const base = { width: "100%", padding: "12px" };
  rules.push(["v-col", { ...base, "flex-basis": "0", "flex-grow": "1", "max-width": "100%" }], ["v-col-auto", { ...base, "flex": "0 0 auto", "width": "auto", "max-width": "100%" }]);
  for (let n = 1; n <= COLS; n++) {
    const pct = colPercent(n);
    rules.push([`v-col-${n}`, { ...base, "flex": `0 0 ${pct}`, "max-width": pct }]);
  }
  for (let n = 1; n < COLS; n++) {
    rules.push([`offset-${n}`, { "margin-inline-start": colPercent(n) }]);
  }
  return rules;
}

function opacityRules() {
  const rules = [];
  for (let i = 0; i <= 100; i += 10) {
    rules.push([`opacity-${i}`, { opacity: String(i / 100) }]);
  }
  return rules;
}

function overflowRules() {
  const rules = [["overflow-auto", { overflow: "auto" }], ["overflow-hidden", { overflow: "hidden" }], ["overflow-visible", { overflow: "visible" }], ["overflow-scroll", { overflow: "scroll" }], ["overflow-x-auto", { "overflow-x": "auto" }], ["overflow-x-hidden", { "overflow-x": "hidden" }], ["overflow-x-visible", { "overflow-x": "visible" }], ["overflow-x-scroll", { "overflow-x": "scroll" }], ["overflow-y-auto", { "overflow-y": "auto" }], ["overflow-y-hidden", { "overflow-y": "hidden" }], ["overflow-y-visible", { "overflow-y": "visible" }], ["overflow-y-scroll", { "overflow-y": "scroll" }]];
  return rules;
}

function pointerEventsRules() {
  const rules = [["pointer-events-none", { "pointer-events": "none" }], ["pointer-events-auto", { "pointer-events": "auto" }]];
  return rules;
}

function positionRules() {
  const rules = [["position-static", { position: "static" }], ["position-relative", { position: "relative" }], ["position-absolute", { position: "absolute" }], ["position-fixed", { position: "fixed" }], ["position-sticky", { position: "sticky" }], ["top-0", { top: "0" }], ["right-0", { right: "0" }], ["bottom-0", { bottom: "0" }], ["left-0", { left: "0" }], ["inset-0", { top: "0", right: "0", bottom: "0", left: "0" }]];
  return rules;
}

function sizingRules() {
  const rules = [["h-auto", { height: "auto" }], ["h-screen", { height: "100vh" }], ["h-screen-dvh", { height: "100dvh" }], ["h-0", { height: "0" }], ["h-25", { height: "25%" }], ["h-50", { height: "50%" }], ["h-75", { height: "75%" }], ["h-100", { height: "100%" }], ["w-auto", { width: "auto" }], ["w-0", { width: "0" }], ["w-25", { width: "25%" }], ["w-33", { width: "33.3333%" }], ["w-50", { width: "50%" }], ["w-66", { width: "66.6667%" }], ["w-75", { width: "75%" }], ["w-100", { width: "100%" }], ["fill-height", { height: "100%" }], ["max-w-100", { "max-width": "100%" }], ["max-h-100", { "max-height": "100%" }], ["min-w-0", { "min-width": "0" }], ["min-h-0", { "min-height": "0" }]];
  return rules;
}

function spacingRules(options) {
  const spacer = options.spacer ?? defaultSpacer;
  const steps = options.spacerSteps ?? defaultSpacerSteps;
  const rules = [[
    /^ga-(\d+)$/,
    ([, n]) => {
      const num = Number.parseInt(n);
      if (num > steps) {
        return;
      }
      return { gap: spacerValue(num, spacer) };
    }
  ], [
    /^gr-(\d+)$/,
    ([, n]) => {
      const num = Number.parseInt(n);
      if (num > steps) {
        return;
      }
      return { "row-gap": spacerValue(num, spacer) };
    }
  ], [
    /^gc-(\d+)$/,
    ([, n]) => {
      const num = Number.parseInt(n);
      if (num > steps) {
        return;
      }
      return { "column-gap": spacerValue(num, spacer) };
    }
  ]];
  for (const [dir, suffixes] of Object.entries(directions)) {
    const prop = suffixes[0] === "" ? "margin" : suffixes.map((s) => `margin${s}`);
    rules.push(
      // Positive values: m{dir}-{0-16}
      [
        new RegExp(String.raw`^m${dir}-(\d+)$`),
        ([, n]) => {
          const num = Number.parseInt(n);
          if (num > steps) {
            return;
          }
          const value = spacerValue(num, spacer);
          if (Array.isArray(prop)) {
            return Object.fromEntries(prop.map((p) => [p, value]));
          }
          return { [prop]: value };
        }
      ],
      // Negative values: m{dir}-n{1-16}
      [
        new RegExp(String.raw`^m${dir}-n(\d+)$`),
        ([, n]) => {
          const num = Number.parseInt(n);
          if (num < 1 || num > steps) {
            return;
          }
          const value = `-${spacerValue(num, spacer)}`;
          if (Array.isArray(prop)) {
            return Object.fromEntries(prop.map((p) => [p, value]));
          }
          return { [prop]: value };
        }
      ],
      // Auto value: m{dir}-auto
      [
        new RegExp(`^m${dir}-auto$`),
        () => {
          if (Array.isArray(prop)) {
            return Object.fromEntries(prop.map((p) => [p, "auto"]));
          }
          return { [prop]: "auto" };
        }
      ]
    );
  }
  for (const [dir, suffixes] of Object.entries(directions)) {
    const prop = suffixes[0] === "" ? "padding" : suffixes.map((s) => `padding${s}`);
    rules.push([
      new RegExp(String.raw`^p${dir}-(\d+)$`),
      ([, n]) => {
        const num = Number.parseInt(n);
        if (num > steps) {
          return;
        }
        const value = spacerValue(num, spacer);
        if (Array.isArray(prop)) {
          return Object.fromEntries(prop.map((p) => [p, value]));
        }
        return { [prop]: value };
      }
    ]);
  }
  return rules;
}

function typographyRules(typography = "md3") {
  const rules = [];
  const baseShortcuts = typography === "md2" ? typographyPresets.md2 : typography === "md3" ? typographyPresets.md3 : typography;
  for (const [name, styles] of Object.entries(baseShortcuts)) {
    const className = name.replace(/([a-z]{2,})(\d)/, "$1-$2");
    const css = Object.entries(styles).reduce(
      (o, [key, value]) => ({ ...o, [kebabCase(key)]: value }),
      {}
    );
    rules.push([`text-${className}`, css, { layer: "typography" }]);
  }
  rules.push(
    ["text-left", { "text-align": "left" }],
    ["text-right", { "text-align": "right" }],
    ["text-center", { "text-align": "center" }],
    ["text-justify", { "text-align": "justify" }],
    ["text-uppercase", { "text-transform": "uppercase" }],
    ["text-lowercase", { "text-transform": "lowercase" }],
    ["text-capitalize", { "text-transform": "capitalize" }],
    ["text-none", { "text-transform": "none" }],
    ["text-decoration-none", { "text-decoration": "none" }],
    ["text-decoration-underline", { "text-decoration": "underline" }],
    ["text-decoration-overline", { "text-decoration": "overline" }],
    ["text-decoration-line-through", { "text-decoration": "line-through" }],
    ["text-truncate", { "overflow": "hidden", "text-overflow": "ellipsis", "white-space": "nowrap" }],
    ["text-no-wrap", { "white-space": "nowrap" }],
    ["text-pre-wrap", { "white-space": "pre-wrap" }],
    ["text-break", { "overflow-wrap": "break-word" }]
  );
  for (const [name, weight] of Object.entries(fontWeights)) {
    rules.push([`font-weight-${name}`, { "font-weight": String(weight) }]);
  }
  for (const weight of [100, 300, 400, 500, 600, 700, 900]) {
    rules.push([`font-weight-${weight}`, { "font-weight": String(weight) }]);
  }
  rules.push(
    ["font-italic", { "font-style": "italic" }],
    ["text-italic", { "font-style": "italic" }],
    ["text-mono", { "font-family": "monospace" }],
    ["text-high-emphasis", { opacity: "var(--v-high-emphasis-opacity, 0.87)" }],
    ["text-medium-emphasis", { opacity: "var(--v-medium-emphasis-opacity, 0.6)" }],
    ["text-disabled", { opacity: "var(--v-disabled-opacity, 0.38)" }]
  );
  return rules;
}

function createRules(options) {
  const exclude = new Set(options.exclude);
  return [
    ...exclude.has("spacing") ? [] : spacingRules(options),
    ...exclude.has("display") ? [] : displayRules(),
    ...exclude.has("flex") ? [] : flexRules(),
    ...exclude.has("typography") ? [] : typographyRules(options.typography),
    ...exclude.has("borders") ? [] : borderRules(),
    ...exclude.has("sizing") ? [] : sizingRules(),
    ...exclude.has("position") ? [] : positionRules(),
    ...exclude.has("overflow") ? [] : overflowRules(),
    ...exclude.has("float") ? [] : floatRules(),
    ...exclude.has("opacity") ? [] : opacityRules(),
    ...exclude.has("cursor") ? [] : cursorRules(),
    ...exclude.has("elevation") ? [] : elevationRules(options.elevation),
    ...exclude.has("pointerEvents") ? [] : pointerEventsRules(),
    ...exclude.has("colors") ? [] : colorRules(),
    ...exclude.has("grid") ? [] : gridRules()
  ].map(([key, styles, options2]) => [key, styles, options2 ?? { layer: "utilities" }]);
}

function breakpointVariants(options, staticRuleNames) {
  const breakpoints = options.breakpoints ?? defaultBreakpoints;
  const variants = [];
  for (const [name, minWidth] of Object.entries(breakpoints)) {
    variants.push({
      name: `vuetify-${name}`,
      match(matcher) {
        if (staticRuleNames.has(matcher)) {
          return;
        }
        const regex = new RegExp(`^(.+)-${name}$`);
        const match = matcher.match(regex);
        if (!match) {
          return;
        }
        return {
          matcher: match[1],
          handle: (input, next) => next({
            ...input,
            parent: `${input.parent ? `${input.parent} $$ ` : ""}@media (min-width: ${minWidth})`
          })
        };
      },
      order: 1e3,
      multiPass: true
    });
  }
  return variants;
}

function printVariant() {
  return {
    name: "vuetify-print",
    match(matcher) {
      if (!matcher.startsWith("d-print-")) {
        return;
      }
      const value = matcher.slice(8);
      return {
        matcher: `d-${value}`,
        handle: (input, next) => next({
          ...input,
          parent: `${input.parent ? `${input.parent} $$ ` : ""}@media print`
        })
      };
    },
    order: 100
  };
}

function createThemeVariants(themes, options) {
  return themes.map((theme) => ({
    order: options ? options.order : -1,
    name: `vuetify-theme-${theme}`,
    match(matcher) {
      const prefix = `${theme}:`;
      if (!matcher.startsWith(prefix)) {
        return;
      }
      return {
        matcher: matcher.slice(prefix.length),
        handle: (input, next) => next({
          ...input,
          parent: `.v-theme--${theme}${input.parent ? ` $$ ${input.parent}` : ""}`
        })
      };
    }
  }));
}

function visibilityVariants(options) {
  const breakpoints = options.breakpoints ?? defaultBreakpoints;
  const breakpointEntries = Object.entries(breakpoints);
  const variants = [];
  for (const [index, [name, min]] of breakpointEntries.entries()) {
    const max = breakpointEntries[index + 1]?.[1];
    variants.push({
      name: `vuetify-hidden-${name}-only`,
      match(matcher) {
        if (matcher !== `hidden-${name}-only`) {
          return;
        }
        const mediaQuery = max ? `(min-width: ${min}) and (max-width: ${Number.parseFloat(max) - 0.02}px)` : `(min-width: ${min})`;
        return {
          matcher: "d-none",
          handle: (input, next) => next({
            ...input,
            parent: `${input.parent ? `${input.parent} $$ ` : ""}@media ${mediaQuery}`
          })
        };
      },
      order: 100
    });
  }
  for (const [index, [name]] of breakpointEntries.entries()) {
    const max = breakpointEntries[index + 1]?.[1];
    variants.push({
      name: `vuetify-hidden-${name}-and-down`,
      match(matcher) {
        if (matcher !== `hidden-${name}-and-down`) {
          return;
        }
        if (max) {
          return {
            matcher: "d-none",
            handle: (input, next) => next({
              ...input,
              parent: `${input.parent ? `${input.parent} $$ ` : ""}@media (max-width: ${Number.parseFloat(max) - 0.02}px)`
            })
          };
        }
        return { matcher: "d-none" };
      },
      order: 100
    });
  }
  for (const [name, min] of breakpointEntries) {
    variants.push({
      name: `vuetify-hidden-${name}-and-up`,
      match(matcher) {
        if (matcher !== `hidden-${name}-and-up`) {
          return;
        }
        return {
          matcher: "d-none",
          handle: (input, next) => next({
            ...input,
            parent: `${input.parent ? `${input.parent} $$ ` : ""}@media (min-width: ${min})`
          })
        };
      },
      order: 100
    });
  }
  variants.push(
    {
      name: "vuetify-hidden-screen-only",
      match(matcher) {
        if (matcher !== "hidden-screen-only") {
          return;
        }
        return {
          matcher: "d-none",
          handle: (input, next) => next({
            ...input,
            parent: `${input.parent ? `${input.parent} $$ ` : ""}@media screen`
          })
        };
      },
      order: 100
    },
    {
      name: "vuetify-hidden-print-only",
      match(matcher) {
        if (matcher !== "hidden-print-only") {
          return;
        }
        return {
          matcher: "d-none",
          handle: (input, next) => next({
            ...input,
            parent: `${input.parent ? `${input.parent} $$ ` : ""}@media print`
          })
        };
      },
      order: 100
    }
  );
  return variants;
}

function createRtlRules(options) {
  const spacer = options.spacer ?? defaultSpacer;
  const steps = options.spacerSteps ?? defaultSpacerSteps;
  const rules = [];
  for (let i = 0; i <= steps; i++) {
    const value = spacerValue(i, spacer);
    rules.push([`ms-${i}`, { "margin-inline-start": value }], [`me-${i}`, { "margin-inline-end": value }]);
  }
  for (let i = 1; i <= steps; i++) {
    const value = `-${spacerValue(i, spacer)}`;
    rules.push([`ms-n${i}`, { "margin-inline-start": value }], [`me-n${i}`, { "margin-inline-end": value }]);
  }
  rules.push(["ms-auto", { "margin-inline-start": "auto" }], ["me-auto", { "margin-inline-end": "auto" }]);
  for (let i = 0; i <= steps; i++) {
    const value = spacerValue(i, spacer);
    rules.push([`ps-${i}`, { "padding-inline-start": value }], [`pe-${i}`, { "padding-inline-end": value }]);
  }
  rules.push(["text-start", { "text-align": "start" }], ["text-end", { "text-align": "end" }], ["float-start", { float: "inline-start" }], ["float-end", { float: "inline-end" }]);
  return rules;
}

function createVariants(options, staticRuleNames) {
  return [
    ...breakpointVariants(options, staticRuleNames),
    ...visibilityVariants(options),
    printVariant(),
    ...options.themes ? createThemeVariants(options.themes) : []
  ];
}

function presetVuetify(options = {}) {
  const rules = [...createRules(options), ...createRtlRules(options)];
  const staticRuleNames = new Set(
    rules.filter((r) => typeof r[0] === "string").map((r) => r[0])
  );
  return {
    name: "unocss-preset-vuetify",
    layers: {
      typography: -1,
      utilities: 0
    },
    rules,
    variants: createVariants(options, staticRuleNames),
    preflights: preflights(options),
    postprocess(util) {
      if (!options.important) {
        return;
      }
      for (const entry of util.entries) {
        if (entry[1] != null && !String(entry[1]).endsWith("!important")) {
          entry[1] = `${entry[1]} !important`;
        }
      }
    }
  };
}

exports.borderRadii = borderRadii;
exports.createThemeVariants = createThemeVariants;
exports.defaultBreakpoints = defaultBreakpoints;
exports.defaultSpacer = defaultSpacer;
exports.defaultSpacerSteps = defaultSpacerSteps;
exports.elevationPresets = elevationPresets;
exports.fontWeights = fontWeights;
exports.presetVuetify = presetVuetify;
exports.typographyPresets = typographyPresets;
